packages/amplify-provider-awscloudformation/src/root-stack-builder/root-stack-builder.ts

```diff
@@ -0,0 +1,252 @@
+import * as cdk from '@aws-cdk/core';
+import * as s3 from '@aws-cdk/aws-s3';
+import * as iam from '@aws-cdk/aws-iam';
+import * as path from 'path';
+import { prepareApp } from '@aws-cdk/core/lib/private/prepare-app';
+import { AmplifyRootStackResourceProps, AmplifyRootStackTemplateProps } from './types';
+import { JSONUtilities, pathManager } from 'amplify-cli-core';
+import { Template } from 'cloudform-types';
+
+const CFN_TEMPLATE_FORMAT_VERSION = '2010-09-09';
+const ROOT_CFN_DESCRIPTION = 'Root Stack for AWS Amplify CLI';
+
+export class AmplifyRootStack extends cdk.Stack {
+  public templateObj: AmplifyRootStackTemplateProps;
+  constructor(scope: cdk.Construct, id: string) {
+    super(scope, id);
+    this.templateOptions.templateFormatVersion = CFN_TEMPLATE_FORMAT_VERSION;
+    this.templateOptions.description = ROOT_CFN_DESCRIPTION;
+    const CfnParameters: { [k: string]: cdk.CfnParameter } = {};
+
+    CfnParameters.deploymentBucketName = new cdk.CfnParameter(this, 'DeploymentBucketName', {
+      type: 'String',
+      description: 'Name of the common deployment bucket provided by the parent stack',
+      default: 'DeploymentBucket',
+    });
+
+    CfnParameters.authRoleName = new cdk.CfnParameter(this, 'AuthRoleName', {
+      type: 'String',
+      default: 'AuthRoleName',
+    });
+
+    CfnParameters.unauthRoleName = new cdk.CfnParameter(this, 'UnauthRoleName', {
+      type: 'String',
+      default: 'UnauthRoleName',
+    });
```

Comment:
Is it necessary to store this in an object. My understanding was that when you create a CfnParameter, it automatically gets added to the stack. Specifically if this needs to be stored in an object

Reply:
I am using templateObj object to be passed to overrideFn so that they can directly modify resources or ant new parameters, mapping , outputs as per their usecase.

This template Obj is now part of RootStackTransform class not the cdk-builder class(root-stack-builder.ts)